plugins/aea-ledger-hwi/aea_ledger_hwi/account.py

```diff
@@ -0,0 +1,356 @@
+        return HWIAccountData(
+            public_key=parsed_response.public_key,
+            address=("0x" + parsed_response.address),
+            chain_code=parsed_response.chain_code if opts.return_chain_code else None,
+        )
+
+    @property
+    def address(self) -> ChecksumAddress:
+        """Address"""
+        if self.default_account is None:
+            self.default_account = self.get_account()
+        return self.default_account.address
+
+    @property
+    def public_key(self) -> ChecksumAddress:
+        """Public key"""
+        if self.default_account is None:
+            self.default_account = self.get_account()
+        return self.default_account.public_key
+
+    def sign_message(
+        self, signable_message: SignableMessage, **kwargs: Any
+    ) -> SignedMessage:
+        """Sign a EIP191 message"""
+
+        raise NotImplementedError()
+
+    @staticmethod
+    def encode_transaction(
+        transaction: TypedTransaction,
+        is_eip1559_tx: bool = False,
+        key_index: Optional[int] = None,
+    ) -> bytes:
+        """Build and encode transaction"""
+
+        # BIP32 Path m/44'/cointype'/account'/change/address
+        path = m / h(44) / h(60) / h(key_index or 0) / 0 / 0
+        path_construct = PrefixedArray(Byte, Int32ub)
+        path_apdu = path_construct.build(path.to_list())
+
+        if is_eip1559_tx:
+            as_dict = transaction.as_dict()
+            tx = UnsignedDynamicTransaction.from_dict(
+                field_dict=dict(
+                    chain_id=as_dict["chainId"],
+                    nonce=as_dict["nonce"],
+                    max_priority_fee_per_gas=as_dict["maxPriorityFeePerGas"],
+                    max_fee_per_gas=as_dict["maxFeePerGas"],
+                    gas_limit=as_dict.get("gas", 21000),
+                    destination=as_dict["to"],
+                    amount=as_dict["value"],
+                    data=as_dict["data"],
+                    access_list=as_dict.get("accessList", []),
+                )
+            )
+            return (
+                path_apdu
+                + transaction.transaction_type.to_bytes(1, "big")
+                + rlp.encode(tx)
+            )
+
+        as_dict = transaction.as_dict()
+        tx = UnsignedTransaction.from_dict(
+            field_dict=dict(
+                nonce=as_dict["nonce"],
+                gasPrice=as_dict["gasPrice"],
+                gas=as_dict["gas"],
+                to=as_dict["to"],
+                value=as_dict["value"],
+                data=as_dict["data"],
+            )
+        )
+        return path_apdu + rlp.encode(tx)
+
+    def _sign_transaction_on_hwi(
+        self,
+        unsigned_transaction: TypedTransaction,
+        key_index: Optional[int] = None,
+        device_index: Optional[int] = None,
+        chain_id: int = 1,
+    ) -> HWISignedTransaction:
+        """Hardware wallet signed transaction"""
+
+        ins = 0x04
+        p2 = 0x00
```

Comment:
What are these. Provide comments in code

Reply:
Extracted them as proper constants